components/pkg-export-kubernetes/src/main.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_use]
+extern crate clap;
+extern crate env_logger;
+extern crate habitat_core as hcore;
+extern crate habitat_common as common;
+extern crate handlebars;
+#[macro_use]
+extern crate serde_json;
+#[macro_use]
+extern crate log;
+
+use clap::App;
+use handlebars::Handlebars;
+use std::result;
+use std::str::FromStr;
+use std::io::prelude::*;
+use std::io;
+use std::fs::File;
+
+use hcore::PROGRAM_NAME;
+use hcore::env as henv;
+use hcore::package::PackageIdent;
+use common::ui::{Coloring, UI, NOCOLORING_ENVVAR, NONINTERACTIVE_ENVVAR};
+
+// Synced with the version of the Habitat operator.
+pub const VERSION: &'static str = "0.1.0";
+
+// Kubernetes manifest template
+const MANIFESTFILE: &'static str = include_str!("../defaults/KubernetesManifest.hbs");
+
+fn main() {
+    env_logger::init().unwrap();
+    let mut ui = ui();
+    if let Err(e) = start(&mut ui) {
+        let _ = ui.fatal(e);
+        std::process::exit(1)
+    }
+}
+
+fn ui() -> UI {
+    let isatty = if henv::var(NONINTERACTIVE_ENVVAR)
+        .map(|val| val == "true")
+        .unwrap_or(false)
+    {
+        Some(false)
+    } else {
+        None
+    };
+    let coloring = if henv::var(NOCOLORING_ENVVAR)
+        .map(|val| val == "true")
+        .unwrap_or(false)
+    {
+        Coloring::Never
+    } else {
+        Coloring::Auto
+    };
+    UI::default_with(coloring, isatty)
+}
+
+fn start(_ui: &mut UI) -> result::Result<(), String> {
+    let m = cli().get_matches();
+    debug!("clap cli args: {:?}", m);
+    let count = m.value_of("COUNT").unwrap_or("1");
+    let topology = m.value_of("TOPOLOGY").unwrap_or("standalone");
+    let (group, group_commentout) = match m.value_of("GROUP") {
+        Some(g) => (g, ""),
+        None => ("default", "#"),
+    };
+    let (config_secret_name, config_secret_name_commentout) =
+        match m.value_of("CONFIG_SECRET_NAME") {
+            Some(s) => (s, ""),
+            None => ("user-toml-secret", "#"),
```

Comment:
The "comment out" string should be `"# "` and not `"#"` (but see earlier comment about flow control in the template)

Reply:
I actually went with "#" but IIRC yamllint didn't like that.

Reply:
I remembered incorrectly, yamllint actually warned about lack of space. The reason now I recall was that I just followed what was already being done by @krnowak and I'm guessing he went for this approach because that's a convention for commenting out directives in config files. If you look at various .conf files under /etc, you'll see this convention being followed by most. However, if you feel strong about it, I'm fine with changing it.